src/bridges.jl

```diff
@@ -30,3 +30,89 @@ function MOI.get(
     func = MOI.get(model, attr, bridge.constraint)
     return MOI.Bridges.adjoint_map_function(typeof(bridge), func)
 end
+
+function MOI.set(
+    model::MOI.ModelLike,
+    attr::DiffOpt.ForwardConstraintFunction,
+    bridge::MOI.Bridges.Constraint.QuadtoSOCBridge{T},
+    diff::MOI.ScalarQuadraticFunction
+) where {T}
+    func = MOI.get(model, MOI.ConstraintFunction(), bridge.soc)
+    index_map = index_map_to_oneto(func)
+    index_map_to_oneto!(index_map, diff)
+    n = length(index_map.var_map)
+    func_array = sparse_array_representation(func, n, index_map)
+    # x'Qx/2 + a'x + β is bridged into
+    # (1, -a'x - β, U*x) in RSOC
+    # where Q = U' * U
+    # `func_array.terms` is `[0; -a'; U]`
+    U = func_array.terms[3:end, :]
+    diff_array = sparse_array_representation(diff, n, index_map)
+    dU = Matrix(diff_array.quadratic_terms)
+    dU = dU_from_dQ!(Matrix(diff_array.quadratic_terms), U)
+    x = Vector{MOI.VariableIndex}(undef, n)
+    for v in keys(index_map.var_map)
+        x[index_map[v].value] = v
+    end
+    diff_aff = MOI.VectorAffineFunction{T}(
+        MOI.VectorAffineTerm{T}[],
+        zeros(T, size(dU, 1) + 2),
+    )
+    for t in diff.affine_terms
+        push!(diff_aff.terms, MOI.VectorAffineTerm(2, MOI.ScalarAffineTerm(-t.coefficient, t.variable)))
+    end
+    diff_aff.constants[2] = -diff.constant
+    for i in axes(dU, 1)
+        for j in axes(dU, 2)
+            if !iszero(dU[i, j])
+                scalar = MOI.ScalarAffineTerm(dU[i, j], x[j])
+                push!(diff_aff.terms, MOI.VectorAffineTerm(2 + i, scalar))
+            end
+        end
+    end
+    MOI.set(model, attr, bridge.soc, diff_aff)
+    return
+end
+
+"""
+    dU_from_dQ!(dQ, U)
+
+Overwrite `dQ` into `dU` the solution of
```

Comment:
the equation is not super clear here, I cannot deduce whether `dQ` is just an in-place output matrix?
In that case this line below
`dd = view(U, 1:i, j)'view(dQ, 1:i, i)`
is strange since it uses the initial value of dQ

Reply:
Does the function also modify U? If so, it should be specified somewhere

Reply:
> In that case this line below
> `dd = view(U, 1:i, j)'view(dQ, 1:i, i)`
> is strange since it uses the initial value of dQ

It uses the part of `dQ` that was already modified

Reply:
I have clarified it in latest commit